src/ol/renderer/canvas/canvasmaprenderer.js

```diff
@@ -193,11 +227,15 @@ ol.renderer.canvas.Map.prototype.renderFrame = function(frameState) {
       continue;
     }
     if (layerRenderer.prepareFrame(frameState, layerState)) {
+      if (layer instanceof ol.layer.Vector && layer.getSource().getWrapX()) {
```

Comment:
Then you won't be able to use any interactions that use the map's vector replay on repeated worlds. As long as we use FeatureOverlay for highlighting, selecting, editing etc, the only alternative solution I could think of would be to use layer replay groups for FeatureOverlay (with the same wrapX setting as the layer's source), but then the overlay wouldn't be on top of the layer stack.